### ISSUE TITLE
antsMultivariateTemplateConstruction2 does not work

### ISSUE THREAD

**Issue by DavidMolitor92** [ANTsX/ANTs#585]
Hi everybody,
I'm trying to create templates from a single MR Scan to use MAGeTbrain. I followed the Instructions under

[](https://github.com/CobraLab/documentation/wiki/ANTs-Multivariate-Template-Construction)

but when i run
./my-antsMultivariateTemplateConstruction2.sh i'll get errors. 
For example a MemoryAllocationError

> Exception caught: 
itk::MemoryAllocationError (0x7f2f44000940)
Location: "unknown" 
File: /home/david/bin/ants/ITKv4-install/include/ITK-4.12/itkImportImageContainer.hxx
Line: 199
Description: Failed to allocate memory for image.

Or in the following 

> Using single precision for computations.
Input scalar image: mT1.nii.gz
Reference image: antsBTPtemplate0.nii.gz
Transform file does not exist: ./antsBTPmT100GenericAffine.mat
Can't read initial transform ./antsBTPmT100GenericAffine.mat
Using single precision for computations.
Input scalar image: rT2_combined.nii.gz
Reference image: antsBTPtemplate1.nii.gz
Transform file does not exist: ./antsBTPmT100GenericAffine.mat
Can't read initial transform ./antsBTPmT100GenericAffine.mat

Maybe someone can help me?
Thanks in Advance.
Greetings David

**Comment by ntustison** [comment]
Have you tried running [these examples](https://github.com/ntustison/TemplateBuildingExample)?

**Comment by cookpa** [comment]
Running the examples is a good start (they are fast).

Can you also share your script and some information on the images? What's the dimension of the input images?

**Comment by DavidMolitor92** [comment]
Oh the link didn't work. I just used the script found [here](https://github.com/CobraLab/documentation/wiki/ANTs-Multivariate-Template-Construction). It's basically the original. 
The images I want to work with are two MR Scans of the same person, one is T1-weighted, the other one T2,
The T2 Image  has 428x256x364 slices as well as the T1-weighted one. They are both threedimensional and in the same space. 
I'm running the examples now. 
Thanks for the quick answers.

**Comment by stnava** [comment]
official documentation

https://github.com/ntustison/TemplateBuildingExample


brian

**Comment by gdevenyi** [comment]
Lol sorry @stnava, seems the people are desperate for documentation, even when they're written for custom internal usage

**Comment by DavidMolitor92** [comment]
Hi, i tried to run the examples! The Faces did work but the BrainSlices didn't. 
I tried all four commands but everytime after reading the ImageSlices the script exits with a error because the template wasn't created! 

> **ERROR (nifti_image_write_hdr_img2): cannot open output file 'BTP/T_template.nii.gz'
Writing output Your template : BTP/T_template was not created. This indicates trouble! You may want to check correctness of your input parameters. Exiting.

I looked into the script but but the input parameters seem okay top me...

**Comment by stnava** [comment]
does the directory BTP exist?

if not, you might create it, then run again.

it's likely that or you lack permissions to write.



brian

**Comment by DavidMolitor92** [comment]
Thanks for the tip, the examples finished without any errors. 
Line 492 of the antsMultivariateTemplateConstruction2.sh is

> MEMORY ='8gb'

I changed that to 4gb right now, maybe that will help.

**Comment by cookpa** [comment]
I think you're on the right track with memory, which is the usual cause of output not appearing. In a cluster environment, the jobs can be killed by the cluster scheduler with no warning or error message. Or it might refuse to run jobs that request too much.

If the examples run, then the problem is almost certainly with memory. You might need to request more, or use fewer cores.

By default, the template scripts use single precision floating point. If your script is locally modified, you might want to verify that's still the case.

**Comment by DavidMolitor92** [comment]
Okay, unfortunately it still doesn't work. Now the script won't even run. 
When i do ./my-antsMultivariateTemplateConstruction.sh it exits with 

> Your initial template : antsBTPtemplate0.nii.gz[0] was not created.  This indicates trouble!  You may want to check correctness of your input parameters. exiting.

again. I thought, it might still be the fault of some permissions so I tried to sudo run the script, which resulted in 

> It seems that the ANTSPATH environment variable is not set. Please add the ANTSPATH
variable. This can be achieved by editing the .bash_profile in the home directory.
Add:
ANTSPATH=/home/yourname/bin/ants/
Or the correct location of the ANTS binaries.
Alternatively, edit this script ( my-antsMultivariateTemplateConstruction2.sh ) to set up this parameter correctly.

But I'm sure that the ANTSPATH is set correctly in the .bashrc file and even when I edited the script it still won't run.

[Here](https://github.com/CobraLab/documentation/wiki/ANTs-Multivariate-Template-Construction) it says to module load ANTs before trying to run the script, but when I try that, the console just tells me that the command "module" couldn't be found. 

Thanks again.
